### DISCUSSION TITLE
GPU decoder acceleration

### DISCUSSION THREAD

**Discussion by twdragon** [ros-drivers/usb_cam#230] [Ideas]
Should we consider the implementation of a hardware-accelerated decoder for H.264 and optionally, H.265?

**Comment by twdragon** [comment]
@knorth55 @flynneva after the release of the new version, I think, we can implement this based on this example: https://github.com/FFmpeg/FFmpeg/blob/master/doc/examples/hw_decode.c

Here it can be the parameter activating the hardware decoder toolchain, like `hardware_accelerated` forking the initialization sequence for software/hardware FFMPEG API. Also, I guess, it is available and fully supported only by more or less recent FFMPEG releases

**Reply by flynneva** [reply]
@twdragon the ros1 and ros2 versions are still quite different. Which "new release" do you mean?

**Reply by twdragon** [reply]
@flynneva the next one after the code merge we all planned before, as was discussed

**Reply by knorth55** [reply]
No one can stop the new feature implementation.
But I think we can move forward to merge `ros1` and `ros2` branch, too.
I'm sorry but I could not make enough time to do that.

**Reply by twdragon** [reply]
@knorth55 @flynneva I think, the ROS code is now separated enough from V4L2 API, so the merge comes with implementing CMake scenarios and parametrization model. @flynneva could you please look at the new intrinsic control configuration model and estimate how laborious it would be to implement it in ROS2?

**Comment by twdragon** [comment]
The development branch with tested hardware accelerated decoder is ready! Preparing a pull request!

UPDATE: #254
